src/Backend/Modules/MailMotor/Command/SaveSettingsHandler.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Backend\Modules\MailMotor\Command;
+
+/*
+ * This file is part of the Fork CMS MailMotor Module from SIESQO.
+ *
+ * For the full copyright and license information, please view the license
+ * file that was distributed with this source code.
+ */
+
+use Common\ModulesSettings;
+
+final class SaveSettingsHandler
+{
+    /**
+     * @var ModulesSettings
+     */
+    protected $modulesSettings;
+
+    /**
+     * SaveSettingsHandler constructor.
+     *
+     * @param ModulesSettings $modulesSettings
+     */
+    public function __construct(
+        ModulesSettings $modulesSettings
+    ) {
+        $this->modulesSettings = $modulesSettings;
+    }
+
+    /**
+     * @param SaveSettings $settings
+     */
+    public function handle(SaveSettings $settings)
+    {
+        // Define module
+        $module = 'MailMotor';
+
+        // set our settings
+        $this->modulesSettings->set($module, 'mail_engine', $settings->mailEngine);
+        $this->modulesSettings->set($module, 'overwrite_interests', $settings->overwriteInterests);
+
+        // mail engine is empty
+        if ($settings->mailEngine == '') {
+            $this->modulesSettings->delete($module, 'api_key');
+            $this->modulesSettings->delete($module, 'list_id');
+        } else {
```

Comment:
inside the if you add return; and then you drop the else and just add it after the if